### PR TITLE
Update SECURITY.md with anza links and names

### PR DESCRIPTION
**Pull request by willhickey**

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -62,12 +62,12 @@ There is no CI available in the private repository so you must build from source
...
 
 ### 5. Ship the patch
-Once the fix is accepted, a member of the solana-labs/security-incident-response group should prepare a single patch file for each affected branch. The commit title for the patch should only contain the advisory id, and not disclose any further details about the incident.
-Copy the patches to https://release.solana.com/ under a subdirectory named after the advisory id (example: https://release.solana.com/GHSA-hx59-f5g4-jghh/v1.4.patch). Contact a member of the solana-labs/admins group if you require access to release.solana.com
+Once the fix is accepted, a member of the anza-xyz/security-incident-response group should prepare a single patch file for each affected branch. The commit title for the patch should only contain the advisory id, and not disclose any further details about the incident.
+Copy the patches to https://release.solana.com/ under a subdirectory named after the advisory id (example: https://release.solana.com/GHSA-hx59-f5g4-jghh/v1.4.patch). Contact a member of the anza-xyz/admins group if you require access to release.solana.com
```
Comment (yihau) [inline SECURITY.md]: have some concerns here: 
1. I think we shouldn't give the write permission of release bucket to any others. maybe we can have another bucket for it 🤔 
2. if we think that's fine, we will need to update it to https://release.anza.xyz. (the line 74 need to be updated as well)

Comment (willhickey) [reply SECURITY.md]: Yeah good call.

@t-nelson Have we been following this guidance on releasing patch files? I've never done it.

Comment (t-nelson) [reply SECURITY.md]: we really only use these instructions for loss of funds. afaik we've never actually hosted a patch file. we just attach it directly to a message in the chat

Comment (willhickey) [reply SECURITY.md]: I ripped out that whole section. It was overly prescriptive and not representative of how we've been doing things. The new language better reflects the judgement calls we make when shipping patches.

**Review by yihau — DISMISSED** [review]
LGTM! just need to check it we have the email alias for security@anza.xyz 📫

**Comment by willhickey** [comment]
> LGTM! just need to check it we have the email alias for [security@anza.xyz](mailto:security@anza.xyz) 📫

Switched it to security@solana.com (there's a Slack conversation in `#comms-internal`)

**Review by yihau — APPROVED** [review]
sorry for missing this one. am I correct that we won't have an email for security@anza.xyz? (fwiw, we have set up an email for maintainers@anza.xyz)

**Review by kubanemil — APPROVED** [review]

**Pull request merged**